sdk/src/instruction_accounts.rs

```diff
@@ -0,0 +1,202 @@
+use std::collections::HashMap;
+
+use solana_program::{instruction::AccountMeta, pubkey::Pubkey};
+
+use crate::{
+    PROGRAM_ID_ACCOUNT_COMPRESSION, PROGRAM_ID_LIGHT_SYSTEM, PROGRAM_ID_NOOP, PROGRAM_ID_SYSTEM,
+};
+
+/// Collection of Light Protocol system accounts which are sent to the program.
+pub struct LightInstructionAccounts {
+    next_index: u8,
+    map: HashMap<Pubkey, u8>,
+}
+
+impl LightInstructionAccounts {
+    pub fn new(
+        fee_payer: &Pubkey,
+        authority: &Pubkey,
+        registered_program_pda: &Pubkey,
+        account_compression_authority: &Pubkey,
```

Comment:
these accounts are always the same, you can hardcode these.
I think you are missing the cpi signer of the invoking program which is also deterministic you can derive it inside of new from the program id.